openedx_external_enrollments/tests/external_enrollments/tests_mit_hz_external_enrollments.py

```diff
@@ -77,3 +83,145 @@ def test_get_user_id(self):
+        """
+        course_settings = {
+            'external_course_target': 'external-course-target',
+        }
+        check_user_subscription_mock.return_value = None
+
+        result = self.base._get_course_home_url(course_settings, {})  # pylint: disable=protected-access
+
+        check_user_subscription_mock.assert_called_once()
+        self.assertEqual(result, course_settings.get('external_course_target'))
+
+    @patch('openedx_external_enrollments.external_enrollments.mit_hz_external_enrollment.ExternalEnrollment')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.'
+        'mit_hz_external_enrollment.MITHzInstanceExternalEnrollment._check_user'
+    )
+    def test_check_user_subscription_with_existing_enrollment(self, check_user_mock, model_mock):
+        """
+        Test _check_user_subscription method with existing enrollment object.
+        """
+        data = {
+            'course_id': 'course_id',
+            'user_email': 'user_email',
+        }
+        enrollment_mock = Mock()
+        enrollment_mock.save.return_value = None
+        model_mock.objects.get.return_value = enrollment_mock
+        check_user_mock.return_value = {}
+
+        self.base._check_user_subscription(data)  # pylint: disable=protected-access
+
+        model_mock.objects.get.assert_called_once_with(
+            controller_name='mit_hz',
+            course_shell=data.get('course_id'),
+            email=data.get('user_email'),
+            meta={},
+        )
+        check_user_mock.assert_called_once_with('setting_value%7Csetting_value%7Cuser_email')
+        enrollment_mock.save.assert_called_once()
+
+    @patch('openedx_external_enrollments.external_enrollments.mit_hz_external_enrollment.ExternalEnrollment')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.'
+        'mit_hz_external_enrollment.MITHzInstanceExternalEnrollment._check_user'
+    )
+    def test_check_user_subscription_with_non_existing_enrollment(self, check_user_mock, model_mock):
+        """
+        Test _check_user_subscription method with a non existing enrollment object.
+        """
+        data = {
+            'course_id': 'course_id',
+            'user_email': 'user_email',
+        }
+        model_mock.objects.get.side_effect = ObjectDoesNotExist
+
+        self.base._check_user_subscription(data)  # pylint: disable=protected-access
+
+        model_mock.objects.get.assert_called_once_with(
+            controller_name='mit_hz',
+            course_shell=data.get('course_id'),
+            email=data.get('user_email'),
+            meta={},
+        )
+        check_user_mock.assert_not_called()
+
+    @patch('openedx_external_enrollments.external_enrollments.mit_hz_external_enrollment.requests.post')
+    @patch('openedx_external_enrollments.external_enrollments.mit_hz_external_enrollment.ExternalEnrollment')
+    def test_execute_post_failed(self, model_mock, post_mock):
```

Comment:
Also, check that "ExternalEnrollment.objects.filter" and "update" are not called. @ivanvgh